docs: add logging and events developer guide

dwmkerr opened the pull request:
## Summary
- Add comprehensive logging and events guide to developer documentation
- Document when to use Kubernetes events vs logging in ARK controllers
- Include code examples for both event recording and logging
- Add kubectl and k9s commands for viewing events and logs

dwmkerr committed:
docs: add logging and events developer guide

Add comprehensive guide covering when to use Kubernetes events vs logging in ARK controllers, with code examples and viewing instructions for kubectl, k9s, and the dashboard.

iamhassaan reviewed (approved):
Looks good.

skokaina commented on docs/content/developer-guide/logging-and-events.mdx:
```diff
@@ -0,0 +1,89 @@
+}
+
+// Event recording in reconciler
+r.Recorder.Event(&query, "Normal", "LLMCallComplete", eventData)
```
we should use variabls or enums for the level and event name and also explain to reuse the right convention "{Resource}{Action}{State}"

dwmkerr replied:
Good call let me update that

dwmkerr committed:
docs: improve event recording examples with constants and naming convention

- Add event level constants to avoid hardcoded strings
- Update event names to follow {Resource}{Action}{State} convention
- Remove unnecessary QueryReconciler struct from example
